src/SpendPermissionManager.sol

```diff
@@ -291,6 +295,22 @@ contract SpendPermissionManager is EIP712 {
         _transferFrom(spendPermission.token, spendPermission.account, spendPermission.spender, value);
     }
 
+    /// @notice Forward native token to recipient.
+    ///
+    /// @dev Reverts if forward amount not equivalent to expected.
+    ///
+    /// @param recipient Account to forward native token to.
+    function forward(address payable recipient) external payable {
+        // check expected forward value matches call value
+        if (_forwards[recipient] != msg.value) revert();
+
+        // forward value to recipient
+        Address.sendValue(recipient, msg.value);
+
+        // clear forwarding value storage
+        delete _forwards[recipient];
```

Comment:
I'm seeing a potential reentrancy concern here - to follow the checks-effects-interactions pattern, I think we should clear out the value in storage first (move line 311 of deleting storage) before the value is sent to the recipient